### PR TITLE
handle repo argument instead of current dir

### PR DESCRIPTION
**Pull request by paulineribeyre**
When using the `--repo` argument, gen3git should use the specified repo to get release notes/create tags instead of the current repository. This fixes the logic by using the variable `repo` (repository requested by the user) instead of `git` (current dir)

Plus a small change to return the release notes dict and make the export to file optional when calling the `main` function from a script instead of using the CLI

### Bug Fixes
- Properly handle --repo argument

### Improvements
- Main function returns the release notes as a dictionary

### CODE REVIEW

File: gen3git.py [hunk 1]
```diff
@@ -261,19 +261,19 @@ def main(args=None):
             print("Cannot find tag: %s" % to_tag)
             return
     else:
-        stop_commit = git.head.commit
+        stop_commit = repo.get_commits().__getitem__(0)  # latest commit
```
Comment (fantix) [inline gen3git.py]: == `repo.get_commits()[0]`?

File: gen3git.py [hunk 2]
```diff
@@ -261,19 +261,19 @@ def main(args=None):
...
         else:
-            for tag in git.tags:
-                if tag.commit.hexsha == stop_commit.hexsha:
+            for tag in repo.get_tags():
+                if tag.commit.sha == stop_commit.sha:
                     stop_tag = tag.name
-    repo.get_commit(stop_commit.hexsha)
-    print("Generate changelog up to commit: %s" % stop_commit.hexsha)
+    repo.get_commit(stop_commit.sha)
+    print("Generate changelog up to commit: %s" % stop_commit.sha)
```
Comment (fantix) [inline gen3git.py:272]: I didn't find doc to `Commit.sha` 🤔  this is the closest: https://gitpython.readthedocs.io/en/stable/reference.html#git.objects.base.Object.binsha

What is it and what is the difference than `hexsha`?

**Review by fantix — APPROVED** [review]

File: gen3git.py (hunk 2 above)
Comment (paulineribeyre) [reply gen3git.py:272]: the `gitpython` lib uses `hexsha`, the `pygithub` lib uses `sha` (https://pygithub.readthedocs.io/en/latest/github_objects/Commit.html#github.Commit.Commit.sha)
the `git` object and the `repo` object are from different libs 🤣

Comment (fantix) [reply gen3git.py:272]: 😱

**Pull request merged**